functions/api/v1/instance.ts

```diff
@@ -1,36 +1,93 @@
+// https://docs.joinmastodon.org/entities/Instance/
+// https://docs.joinmastodon.org/methods/instance/
 import type { Env } from 'wildebeest/backend/src/types/env'
 import { cors } from 'wildebeest/backend/src/utils/cors'
+import * as error from 'wildebeest/backend/src/errors'
 import { DEFAULT_THUMBNAIL } from 'wildebeest/backend/src/config'
 import { getVersion } from 'wildebeest/config/versions'
+import { calculateInstanceStatistics } from 'wildebeest/backend/src/mastodon/instance'
+import { MastodonInstance, InstanceStatistics } from 'wildebeest/backend/src/types/instance'
+import { MastodonAccount } from 'wildebeest/backend/src/types/account'
+import { loadLocalMastodonAccount } from 'wildebeest/backend/src/mastodon/account'
+import { Database, getDatabase } from 'wildebeest/backend/src/database'
+import { getAdmins } from 'wildebeest/backend/src/utils/auth/getAdmins'
+import { emailSymbol } from 'wildebeest/backend/src/activitypub/actors'
 
 export const onRequest: PagesFunction<Env, any> = async ({ env, request }) => {
-	const domain = new URL(request.url).hostname
-	return handleRequest(domain, env)
+	const domain: string = new URL(request.url).hostname
+	const dbOverride: Database = await getDatabase(env)
+	return handleRequest(domain, env, dbOverride)
 }
 
-export async function handleRequest(domain: string, env: Env) {
+export async function handleRequest(domain: string, env: Env, dbOverride?: Database) {
 	const headers = {
 		...cors(),
 		'content-type': 'application/json; charset=utf-8',
 	}
 
-	const res: any = {}
-
-	res.thumbnail = DEFAULT_THUMBNAIL
-
-	// Registration is disabled because unsupported by Wildebeest. Users
-	// should go through the login flow and authenticate with Access.
-	// The documentation is incorrect and registrations is a boolean.
-	res.registrations = false
+	const res: MastodonInstance = {
+		uri: domain,
+		title: env?.INSTANCE_TITLE,
+		description: env?.INSTANCE_DESCR,
+		short_description: env?.INSTANCE_DESCR,
+		email: env?.ADMIN_EMAIL,
+		version: getVersion(domain),
+		languages: ['en'],
+		registrations: env?.INSTANCE_ACCEPTING_REGISTRATIONS ?? false,
+		approval_required: env?.INSTANCE_REGISTRATIONS_REQUIRE_APPROVAL ?? false,
+		invites_enabled: false,
+		thumbnail: DEFAULT_THUMBNAIL,
+		configuration: {
+			statuses: {
+				max_characters: env?.INSTANCE_CONFIG_STATUSES_MAX_CHARACTERS ?? 500,
+				max_media_attachments: 4,
+				characters_reserved_per_url: 23,
+			},
+			media_attachments: {
+				supported_mime_types: ['image/jpeg', 'image/png', 'image/gif', 'image/webp', 'video/mp4'],
+				image_size_limit: 10485760,
+				image_matrix_limit: 16777216,
+				video_size_limit: 41943040,
+				video_frame_rate_limit: 60,
+				video_matrix_limit: 2304000,
+			},
+			polls: {
```

Comment:
We don't support polls at the moment

Reply:
Some Mastodon clients will throw indecipherable errors if they don't receive a valid object for the `polls` property. I changed the parameters to get beyond typechecking validation steps in such clients, while also making it clear that polls are not yet supported.